sklearn/impute.py

```diff
@@ -551,13 +604,15 @@ def fit(self, X, y=None):
         self : object
             Returns self.
         """
+        self.fully_missing_indexes_ = None
         if not is_scalar_nan(self.missing_values):
             force_all_finite = True
         else:
             force_all_finite = "allow-nan"
         X = check_array(X, accept_sparse=('csc', 'csr'),
                         force_all_finite=force_all_finite)
         _check_inputs_dtype(X, self.missing_values)
+        X = self._drop_constant_columns(X)
```

Comment:
Doing this by default breaks backwards compatibility. This is a serious issue with the PR as it stands.

I agree that this is sensible behaviour, but it is also handling a rare edge case and I would rather waste little code on it. My main concern for a column with all NaNs is that we do not raise an obscure error message. I don't really care how it's handled as long as it's backwards compatible.